vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/FieldValidationUtil.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.checkbox;
+
+class FieldValidationUtil {
+    private FieldValidationUtil() {
+    }
+
+    static <T> void disableClientValidation(CheckboxGroup<T> component) {
+        // Since this method should be called for every time when the component
+        // is attached to the UI, lets check that it is actually so
+        if (!component.isAttached()) {
+            throw new IllegalStateException(String.format(
+                    "Component %s is not attached. Client side validation "
+                            + "should be disabled when the component is "
+                            + "attached, thus this method needs to be called "
+                            + "from the onAttach method of the component.",
+                    component));
+        }
+        // Wait until the response is being written as the validation state
+        // should not change after that
+        component.getUI().ifPresent(ui -> ui.beforeClientResponse(component,
+                executionContext -> overrideClientValidation(component)));
```

Comment:
Looking at other similar places where the client-side validation is getting disabled for other components, it seems that they do it slightly differently:

```java
final StateNode componentNode = component.getElement().getNode();
componentNode.runWhenAttached(ui -> ui.getInternals().getStateTree()
        .beforeClientResponse(componentNode,
                executionContext -> overrideClientValidation(
                        component)));
```

Just want to make sure if there is any significant difference.

Reply:
Right, I simplified that bit, because the `runWhenAttached` wasn't necessary, as the assertion above checks that the component is already attached. Apart from that accessing the `StateNode` is not necessary, and neither is accessing the internal state tree to use `beforeClientResponse`.

Reply:
OK 👍 It would be good to simplify that in other components at some point as well.